### PR TITLE
GS-95

### PR DESCRIPTION
**Pull request by cshields-path**
Adds CyUSB Tools:

Installs the following tools alongside the Cypress drivers:
- `lscygpio`
- `cygpioget`
- `cygpioset`
- `cygpio-hammer`
### Other changes
- Update readme to give usage instructions
- Increment the minor version

### CODE REVIEW

**Review by lkucalaba — APPROVED** [review]

File: modules/cyusb-tools/src/cygpioset.cpp [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    {
+      case 0:  // long option
+      {
+        const char* p_cstr_long_argument_name = a_long_options[option_index].name;
+        DEBUGLOG("[DEBUG] Option " << p_cstr_long_argument_name << " was provided");
+        if (::optarg == nullptr)
+        {
+          DEBUGLOG(std::endl);
+          /*
+          if (std::strcmp(p_cstr_long_argument_name, "help") == 0)
```
Comment (bruane-path) [inline modules/cyusb-tools/src/cygpioset.cpp:74]: no longer needed?

File: modules/cyusb-tools/src/cygpioset.cpp [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+        ++regular_arg_index;
+      }
+      break;
+
+      case 'h':
+        DEBUGLOG("[DEBUG] Option h was provided" << std::endl);
+        enum_exit_code = ExitCodeEnum::PRINT_USAGE_AND_EXIT;
+        break;
+
+        /*
```
Comment (bruane-path) [inline modules/cyusb-tools/src/cygpioset.cpp:146]: Are these potential future features?

**Review by bruane-path — APPROVED** [review]
LGTM.  Saw it in action when Cory was testing it and it was solid / robust.  Nice work @lkucalaba !

File: modules/cyusb-tools/src/cygpioset.cpp (hunk 1 above)
Comment (lkucalaba) [reply modules/cyusb-tools/src/cygpioset.cpp:74]: This is demonstrative comments on how to parse long form options if they are unique.  I learned later that you can map long options to short options with the 'val' field of the 'option' struct--so in this case, the --help long option will invoke the -h short option handler (opt='h' in the switch/case).  I'm fine with removing this but note if we add long options then we'll have to use this pattern again

File: modules/cyusb-tools/src/cygpioset.cpp (hunk 2 above)
Comment (lkucalaba) [reply modules/cyusb-tools/src/cygpioset.cpp:146]: These are demonstrative/testing comments and logging for features that I was going add (like -m, -s, -u) but i ran out of time.  This pattern is for parsing simple string args, integer args, or just bool optoin flags.  Fine with removing, but then if we add later we'll have to figure this pattern out again

**Pull request merged**